documentation/puppet-api/v4/compile.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Compile"
+canonical: "/puppetserver/latest/puppet-api/v4/compile.html"
+---
+
+The compile endpoint allows compilation of arbitrary pcore serialized Puppet
+Code ASTs (aka parsed, but unevaluated Puppet Code snippets). In addition to
+the code AST, the caller must provide the certname, environment, facts,
+trusted facts, and variables for compilation (the context in which this AST
+will be evaluated). Optionally, the caller may provide job_id, transaction_id,
+and configure returned logging. The request body must be JSON formatted and
+the caller must accept a JSON response. The server must not have rich_data
+disabled.
+
+
+## `POST /puppet/v4/compile`
+
+(Introduced in Puppet Server 6.5.0)
+
+The request body must look like:
+```
+{
+  "code_ast": "<pcore serialized ast",
+  "certname": "<node name>",
+  "environment": "<environment name>",
+  "facts": { "values": { "<fact name>": <fact value>, ... } },
+  "trusted_facts": { "values": { "<fact name>": <fact value>, ... } },
+  "variables": { "values": { "<variable name>": <variable value>, ... } },
+
+  # The rest are optional:
+  "transaction_uuid": "<uuid string>",
+  "job_id": "<id string>",
+  "options": { "capture_logs": <true/false>,
+               "log_level": <one of debug/info/warn>}
+}
+```
+
+#### `code_ast` (required)
+A parsed string of json encoded
+[PCore Generic Data](https://github.com/puppetlabs/puppet-specifications/blob/a87c9967ebd7071e0a548920f3262ac24cc3f4cd/language/data-types/pcore-generic-data.md)
```

Comment:
I'm not a 100% on the correctness or value of the links in these sections, though I'm not sure what to put in here that would be helpful documentation without going on a code safari. Maybe, since the consumer of this is internal and already doing what they need to do we can get by with a cursory blurb that an AST string goes here?